Subject: [PATCH] UnifiedEmail: fix IllegalStateException

An IllgalStateException occured during monkey test due to committing
Fragment transaction after onSaveInstanceState(). The offending code
is indeed executed on UI thread, so either the click event is fired
after onPause(), or onSaveIntanceState() is called by framework before
onPause(). Use commitAllowingStateLoss() to avoid crash. This won't
affect user because it doesn't make sense anyway to show DialogFragment
when the Activity is not in foreground.

diff --git a/src/com/android/mail/compose/ComposeActivity.java b/src/com/android/mail/compose/ComposeActivity.java
index 5f125ab..db339f5 100644
--- a/src/com/android/mail/compose/ComposeActivity.java
+++ b/src/com/android/mail/compose/ComposeActivity.java
@@ -2872,7 +2872,8 @@
      */
     private void showRecipientErrorDialog(final String message) {
         final DialogFragment frag = RecipientErrorDialogFragment.newInstance(message);
-        frag.show(getFragmentManager(), "recipient error");
+        getFragmentManager().beginTransaction().add(frag, "recipient error")
+                .commitAllowingStateLoss();
     }
 
     /**
